core/services/postgres/transaction.go

```diff
@@ -89,3 +91,75 @@ func DBWithDefaultContext(db *gorm.DB, fc func(db *gorm.DB) error) error {
 	defer cancel()
 	return fc(db.WithContext(ctx))
 }
+
+func WrapDbWithSqlx(rdb *sql.DB) *sqlx.DB {
+	db := sqlx.NewDb(rdb, "postgres")
+	db.MapperFunc(CamelToSnakeASCII)
+	return db
+}
+
+func SqlxTransaction(ctx context.Context, rdb *sql.DB, fc func(tx *sqlx.Tx) error, txOpts ...sql.TxOptions) (err error) {
+	opts := &DefaultSqlTxOptions
+	if len(txOpts) > 0 {
+		opts = &txOpts[0]
+	}
+	db := WrapDbWithSqlx(rdb)
+
+	tx, err := db.BeginTxx(ctx, opts)
+	panicked := false
+
+	defer func() {
+		// Make sure to rollback when panic, Block error or Commit error
+		if panicked || err != nil {
+			if perr := tx.Rollback(); perr != nil {
+				panic(perr)
+			}
+		}
+	}()
+
+	_, err = tx.Exec(fmt.Sprintf(`SET LOCAL lock_timeout = %v; SET LOCAL idle_in_transaction_session_timeout = %v;`, LockTimeout.Milliseconds(), IdleInTxSessionTimeout.Milliseconds()))
+	if err != nil {
+		return errors.Wrap(err, "error setting transaction timeouts")
+	}
+
+	panicked = true
+	err = fc(tx)
+	panicked = false
+
+	if err == nil {
+		err = tx.Commit()
+	}
+
+	return
+}
+
+func allowedBindRune(b byte) bool {
+	return (b >= 'a' && b <= 'z') || (b >= 'A' && b <= 'Z') || (b >= '0' && b <= '9')
+}
+
+// CamelToSnakeASCII converts camel case strings to snake case. For performance
+// reasons it only works with ASCII strings.
+func CamelToSnakeASCII(s string) string {
```

Comment:
Slightly sad that this wasnt available in any stdlib utils

Reply:
Agreed, I copied it from https://github.com/scylladb/go-reflectx 

I'll bring over the test too